### DISCUSSION TITLE
How to let multiple models share one counter?

### DISCUSSION THREAD

**Discussion by Yiak** [typegoose/auto-increment#15] [Q&A] [answered]
I have two models and would like them to share the same topic_id counter.  

```ts
@plugin(AutoIncrementID, {
	field: "topic_id",
	trackerCollection: "meta",
	trackerModelName: "Topic",
	startAt: 100
})
export class Topic{
        @prop({ type: Number, index: true, unique: true })
	public topic_id!: number;
        ...
}
```

```ts
export class Message extends Topic {
	@prop({ type: Boolean, default: true, required: true })
	public is_private!: boolean;
}
```

With the above code, I still got two docs in the meta collection:

<img width="376" alt="1" src="https://user-images.githubusercontent.com/25087436/182040887-bdce79e9-36ad-43b8-848e-b1d286307a85.png">

I thought `trackerModelName` should do the trick but it didn't. So how can I make it work?  Any suggestions would be appreciated.

**Comment by hasezoey** [comment, marked as answer]
> I thought trackerModelName should do the trick but it didn't. So how can I make it work? Any suggestions would be appreciated.

`trackerModelName` sets the name for the model that stores the counter, not the `modelName` property (this is what the documentation is trying to say, but may be not enough on second looks)
as a side note, all properties that start with `tracker` affect the tracker schema / model itself, not the data inside

currently it is not possible to share a counter for multiple models (though on request i may look into adding such a option to overwrite the modelName)

**Reply by Yiak** [reply]
Thank you!

**Reply by hasezoey** [reply]
added the option in bb22a5fda171e36bf8d75defe751a72b4bc17caa, named `overwriteModelName`, this will be available in 1.6.0, which should release after the build is finished (~10min)

**Reply by Yiak** [reply]
Great! Tested it and it works perfectly. Thanks for the work!
